database/entity/0069-add_dlt_message_id/Transaction.ts

```diff
@@ -0,0 +1,155 @@
+import { Decimal } from 'decimal.js-light'
+import { BaseEntity, Entity, PrimaryGeneratedColumn, Column, OneToOne, JoinColumn } from 'typeorm'
+import { DecimalTransformer } from '../../src/typeorm/DecimalTransformer'
+import { Contribution } from '../Contribution'
+
+@Entity('transactions')
+export class Transaction extends BaseEntity {
+  @PrimaryGeneratedColumn('increment', { unsigned: true })
+  id: number
+
+  @Column({ type: 'int', unsigned: true, unique: true, nullable: true, default: null })
+  previous: number | null
+
+  @Column({ name: 'type_id', unsigned: true, nullable: false })
+  typeId: number
+
+  @Column({
+    name: 'transaction_link_id',
+    type: 'int',
+    unsigned: true,
+    nullable: true,
+    default: null,
+  })
+  transactionLinkId?: number | null
+
+  @Column({
+    type: 'decimal',
+    precision: 40,
+    scale: 20,
+    nullable: false,
+    transformer: DecimalTransformer,
+  })
+  amount: Decimal
+
+  @Column({
+    type: 'decimal',
+    precision: 40,
+    scale: 20,
+    nullable: false,
+    transformer: DecimalTransformer,
+  })
+  balance: Decimal
+
+  @Column({
+    name: 'balance_date',
+    type: 'datetime',
+    default: () => 'CURRENT_TIMESTAMP',
+    nullable: false,
+  })
+  balanceDate: Date
+
+  @Column({
+    type: 'decimal',
+    precision: 40,
+    scale: 20,
+    nullable: false,
+    transformer: DecimalTransformer,
+  })
+  decay: Decimal
+
+  @Column({
+    name: 'decay_start',
+    type: 'datetime',
+    nullable: true,
+    default: null,
+  })
+  decayStart: Date | null
+
+  @Column({ length: 255, nullable: false, collation: 'utf8mb4_unicode_ci' })
+  memo: string
+
+  @Column({ name: 'creation_date', type: 'datetime', nullable: true, default: null })
+  creationDate: Date | null
+
+  @Column({ name: 'user_id', unsigned: true, nullable: false })
+  userId: number
+
+  @Column({
+    name: 'user_gradido_id',
+    type: 'varchar',
+    length: 36,
+    nullable: false,
+    collation: 'utf8mb4_unicode_ci',
+  })
+  userGradidoID: string
+
+  @Column({
+    name: 'user_name',
+    type: 'varchar',
+    length: 512,
+    nullable: true,
+    collation: 'utf8mb4_unicode_ci',
+  })
+  userName: string | null
+
+  @Column({
+    name: 'linked_user_id',
+    type: 'int',
+    unsigned: true,
+    nullable: true,
+    default: null,
+  })
+  linkedUserId?: number | null
+
+  @Column({
+    name: 'linked_user_gradido_id',
+    type: 'varchar',
+    length: 36,
+    nullable: true,
+    collation: 'utf8mb4_unicode_ci',
+  })
+  linkedUserGradidoID: string | null
+
+  @Column({
+    name: 'linked_user_name',
+    type: 'varchar',
+    length: 512,
+    nullable: true,
+    collation: 'utf8mb4_unicode_ci',
+  })
+  linkedUserName: string | null
+
+  @Column({
+    name: 'linked_transaction_id',
+    type: 'int',
+    unsigned: true,
+    nullable: true,
+    default: null,
+  })
+  linkedTransactionId?: number | null
+
+  @Column({
+    name: 'dlt_transaction_id',
```

Comment:
whats the business background of this id?
Is it a foreign key to somewhere?
What could the application do with this information later?

So i gues this will cause some conceptuell discussions

Reply:
It is for the iota message id. But as we have named the iota-connector dlt-connector I have also named this dlt_transaction_id rather than iota_transaction_id or iota_message_id.
The iota message id is the easiest way for checking if the transaction is on the tangle, or on a chronicle.
Currently I also plan to use this id for notification the backend if the transaction was 'confirmed' on the tangle.